aci/resource_aci_bfdmhifp.go

```diff
@@ -0,0 +1,305 @@
+	return &schema.Resource{
+		CreateContext: resourceAciBfdMultihopInterfaceProfileCreate,
+		UpdateContext: resourceAciBfdMultihopInterfaceProfileUpdate,
+		ReadContext:   resourceAciBfdMultihopInterfaceProfileRead,
+		DeleteContext: resourceAciBfdMultihopInterfaceProfileDelete,
+
+		Importer: &schema.ResourceImporter{
+			State: resourceAciBfdMultihopInterfaceProfileImport,
+		},
+
+		SchemaVersion: 1,
+		Schema: AppendBaseAttrSchema(AppendNameAliasAttrSchema(map[string]*schema.Schema{
+			"logical_interface_profile_dn": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"key": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"key_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+			},
+			"interface_profile_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+				Computed: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"none",
+					"sha1",
+				}, false),
+			},
+			"relation_bfd_rs_mh_if_pol": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Create relation to bfd:MhIfPol",
+			}})),
+	}
+}
+
+func getRemoteAciBfdMultihopInterfaceProfile(client *client.Client, dn string) (*models.AciBfdMultihopInterfaceProfile, error) {
+	bfdMhIfPCont, err := client.Get(dn)
+	if err != nil {
+		return nil, err
+	}
+	bfdMhIfP := models.AciBfdMultihopInterfaceProfileFromContainer(bfdMhIfPCont)
+	if bfdMhIfP.DistinguishedName == "" {
+		return nil, fmt.Errorf("Aci BFD Multihop Interface Profile %s not found", bfdMhIfP.DistinguishedName)
+	}
+	return bfdMhIfP, nil
+}
+
+func setAciBfdMultihopInterfaceProfileAttributes(bfdMhIfP *models.AciBfdMultihopInterfaceProfile, d *schema.ResourceData) (*schema.ResourceData, error) {
+	dn := d.Id()
+	d.SetId(bfdMhIfP.DistinguishedName)
+	d.Set("description", bfdMhIfP.Description)
+	if dn != bfdMhIfP.DistinguishedName {
+		d.Set("logical_interface_profile_dn", "")
+	}
+	bfdMhIfPMap, err := bfdMhIfP.ToMap()
+	if err != nil {
+		return d, err
+	}
+	d.Set("logical_interface_profile_dn", GetParentDn(dn, fmt.Sprintf("/bfdMhIfP")))
+	d.Set("annotation", bfdMhIfPMap["annotation"])
+	d.Set("key", bfdMhIfPMap["key"])
+	d.Set("key_id", bfdMhIfPMap["keyId"])
+	d.Set("name", bfdMhIfPMap["name"])
+	d.Set("interface_profile_type", bfdMhIfPMap["InterfaceProfile_type"])
+	d.Set("name_alias", bfdMhIfPMap["nameAlias"])
+	return d, nil
+}
+
+func resourceAciBfdMultihopInterfaceProfileImport(d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
+	log.Printf("[DEBUG] Aci BFD Multihop Interface Profile %s: Beginning Import", d.Id())
+	aciClient := m.(*client.Client)
+	dn := d.Id()
+	bfdMhIfP, err := getRemoteAciBfdMultihopInterfaceProfile(aciClient, dn)
```

Comment:
do we need this dn assignment?
```suggestion
	bfdMhIfP, err := getRemoteAciBfdMultihopInterfaceProfile(aciClient, d.Id())
```